src/guides/relations.md

```diff
@@ -0,0 +1,562 @@
+
+```rust
+use diesel::pg::PgConnection;
+use diesel::prelude::*;
+use dotenvy::dotenv;
+use std::env;
+
+pub mod model;
+pub mod schema;
+
+use crate::model::*;
+use crate::schema::*;
+
+pub fn establish_connection() -> PgConnection {
+    dotenv().ok();
+
+    let database_url = env::var("DATABASE_URL").expect("DATABASE_URL must be set");
+    PgConnection::establish(&database_url)
+        .unwrap_or_else(|_| panic!("Error connecting to {}", database_url))
+}
+
+fn main() {
+    let conn = &mut establish_connection();
+
+    let new_book = NewBook {
+        title: "Momo",
+    };
+    let book: Book = diesel::insert_into(books::table)
+        .values(&new_book)
+        .get_result(conn)
+        .expect("Error saving book");
+
+    let new_page_1 = NewPage {
+        page_number: 1,
+        content: "In alten, alten Zeiten ...",
+        book_id: book.id,
+    };
+    let page_1: Page = diesel::insert_into(pages::table)
+        .values(&new_page_1)
+        .get_result(conn)
+        .expect("Error saving page 1");
+
+    let new_page_2 = NewPage {
+        page_number: 2,
+        content: "den prachtvollen Theatern...",
+        book_id: book.id,
+    };
+    let page_2: Page = diesel::insert_into(pages::table)
+        .values(&new_page_2)
+        .get_result(conn)
+        .expect("Error saving page 2");
+
+    // get pages for a book
+    let pages = Page::belonging_to(&book)
+        .inner_join(books::table)
+        .select(pages::all_columns)
+        .load::<Page>(conn)
+        .expect("Error loading pages");
+
+    // the data is the same we put in
+    assert_eq!(&page_1, pages.get(0).unwrap());
+    assert_eq!(&page_2, pages.get(1).unwrap());
+}
+```
+
+:::
+
+## many-to-many or m:n
+
+We currently have books and they have pages, but books have an author, too. To
+be precise, a book can have more than one author and an author can have more
+than one book. So we have a many-to-many relation here. Diesel does not have a
+concept of `has_many` but instead we are going create a "join table"
+`books_authors` and add a `belongs_to` to both `books` and `authors`. Let's see
+that in action.
+
+## Migrations
+
+```sh
+diesel migration generate create_authors
+diesel migration generate create_books_authors
+```
+
+Let's keep it simple and just give our `authors` a name.
+
+::: code-block
+
+[up.sql]( https://github.com/fix-me/up.sql )
+
+```sql
+CREATE TABLE authors (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR NOT NULL,
+);
+```
+:::
+
+And the corresponding down migration.
+
+::: code-block
+
+[down.sql]( https://github.com/fix-me/up.sql )
+
+```sql
+DROP TABLE authors;
+```
+:::
+
+The `books_authors` migration has two foreign keys which point to book's id and
+the author's id. This allows us to create a many-to-many relation between books
+and authors.
+
+::: code-block
+
+[up.sql]( https://github.com/fix-me/up.sql )
+
+```sql
+CREATE TABLE books_authors (
+  id SERIAL PRIMARY KEY,
```

Comment:
The additional `id` column here is not needed. We can just use `(book_id, author_id)` as primary key.

Reply:
If we do, we need to tell diesel what the key is, right? Might be a good place to then link to the documentation and explain how to change it, as you mentioned above.

Reply:
Yes [for `#[derive(Identifiable)]` we need to use `#[diesel(primary_key(book_id, author_id))]`](https://docs.diesel.rs/master/diesel/prelude/derive.Identifiable.html#optional-container-attributes). That's for the primary key although. (The foreign keys don't change in this case)